wasi-tests/src/test/java/com/dylibso/chicory/wasi/WasiPreview1Test.java

```diff
@@ -443,4 +446,53 @@ private static WasiPreview1 wasiWithDirectory(String guest, Path host) {
         var options = WasiOptions.builder().withDirectory(guest, host).build();
         return WasiPreview1.builder().withOptions(options).build();
     }
+
+    // TODO: clean this up
+    // starts, ends but does nothing, need to dig down!
+    @Test
+    @Timeout(value = 30, unit = MINUTES)
+    public void runProtoc() throws Exception {
+        try (var fs = newZeroFs()) {
+            var dir = "protos";
+            Path source = new File("./src/test/resources/protoc-test").toPath().resolve(dir);
+            Path target = fs.getPath("/");
+            copyDirectory(source, target);
+
+            try (var wasi =
+                    WasiPreview1.builder()
+                            .withOptions(
+                                    WasiOptions.builder()
+                                            .inheritSystem()
+                                            .withArguments(
+                                                    List.of(
+                                                            "protoc-gen-java",
+                                                            "-grpc-java_out=out/grpc-java",
+                                                            "-Iprotos helloworld.proto"))
+                                            .withDirectory(target.toString(), target)
+                                            .build())
+                            .build()) {
+                var imports =
+                        ImportValues.builder()
+                                .addFunction(wasi.toHostFunctions())
+                                .addMemory(
+                                        new ImportMemory(
+                                                "env",
+                                                "memory",
+                                                new ByteArrayMemory(
+                                                        new MemoryLimits(
+                                                                /* 7 */ 4,
+                                                                MemoryLimits.MAX_PAGES,
+                                                                true))))
+                                .build();
+                var module =
+                        Parser.parse(
+                                new File(
+                                        "/home/andreatp/workspace/go-protoc-gen-grpc-java/internal/wasm/protoc-gen-grpc-java.wasm"));
```

Comment:
My goal was to run this module, coming from:
https://github.com/wasilibs/go-protoc-gen-grpc-java

But something is not working yet :cry:

Reply:
Working now! Both protoc and jq